# QTL_sticklebacks_backcross
The R scripts used for the QTL analyses on the backcross Pungitius sticklebacks. Please cite these codes using DOI: 10.5281/zenodo.11559306. The related study "Heterogeneous genomic architecture of skeletal armour traits in sticklebacks" will be published in the Journal of Evolutionary Biology (link to add).

The linkage maps were constructed using LepMAP3 (Rastas 2017). All QTL mapping was conducted using the R package R/qtl2 (Broman et al 2019). The Lasso regression method was modified from Kemppainen et al. (2021) thanks to the help of Dr. Kemppainen (https://github.com/petrikemppainen).

Input data and files for these analyses can be found in the Dryad archive for this work (link to add).
  
Citations

Broman, K. W., Gatti, D. M., Simecek, P., Furlotte, N. A., Prins, P., Sen, Ś., ... & Churchill, G. A. (2019). R/qtl2: software for mapping quantitative trait loci with high-dimensional data and multiparent populations. Genetics, 211(2), 495-502.

Kemppainen, P., Li, Z., Rastas, P., Löytynoja, A., Fang, B., Yang, J., ... & Merilä, J. (2021). Genetic population structure constrains local adaptation in sticklebacks. Molecular Ecology, 30(9), 1946-1961. 

Rastas, P. (2017). Lep-MAP3: robust linkage mapping even for low-coverage whole genome sequencing data. Bioinformatics, 33(23), 3726-3732.

